### ISSUE TITLE
GPU memory leaks

### ISSUE THREAD

**Issue by semjon00** [thygate/stable-diffusion-webui-depthmap-script#315]
If generating OOMs, ironically, it does not free memory (standalone mode).

**Comment by T0biasCZe** [comment]
> Batching from directory loads all the images at once. Not ideal.

its very not ideal, because it can hit C runtime max open files limit
And when you have just 1 more file than 8192, it throws `OSError: [Errno 24] Too many open files: './models/leres/res101.pth' `error 
Which is very annoying when trying to do a whole video, since you have to convert it in like 4 rus...

**Comment by semjon00** [comment]
@T0biasCZe Please try out the new Single Video mode :)

**Comment by T0biasCZe** [comment]
>  By the way, 8192 is a lot of files O_O

Wellllll, at 24fps thats just 5 minutes 41seconds :D and considering it runs at 1.5files/s (with res101) it doesnt take that long
And

**Comment by T0biasCZe** [comment]
Also, with the video mode, if the conversion stops mid generation (for example error or something), the already generated images will propably be lost and it will be uncontinuable (which is issue for 20 minute vids...)

**Comment by T0biasCZe** [comment]
Yeah like expected, mid conversion threw error, video gon 😄 
![image](https://github.com/thygate/stable-diffusion-webui-depthmap-script/assets/44525446/64bfca45-d8fd-4e5a-ad2e-cee7c938b890)

**Comment by semjon00** [comment]
> Unless it could be made to output the converted frames into jpegs instead of back into another video thinkingAnd then when converting the frames, it would check if the frame already exists in the folder and potencially skip it thinking

This would make it really complicated. I will think about this, maybe there will be a solution in the future.

**Comment by semjon00** [comment]
Please update the extension and try again.

**Comment by semjon00** [comment]
@T0biasCZe Please update the extension and try again.

**Comment by semjon00** [comment]
No, no. I meant that images that will consist the video are already created - 16-bit grayscale png-s for depthmaps and/or 24-bit images for colorvideos. What was left is to bundle them into a video. Please try again (after updating the extension), it should just work.

**Comment by T0biasCZe** [comment]
Jpeg/Png doesnt matter. And i meant the already processed and finished images, not the raw images from the input video...

sooo... It should be possible to make the processing continue in case it for any reasons stopped during the conversion 🙃

**Comment by T0biasCZe** [comment]
And yeah the new update forks.... *(but that wont help with other mid process interuptions)*

**Comment by semjon00** [comment]
Please send the stacktrace... You know, otherwise I can't help.

**Comment by semjon00** [comment]
> sooo... It should be possible to make the processing continue in case it for any reasons stopped during the conversion upside_down_face

Yep, a good idea, but not priority right now. There are some more pressing issues at the moment... If you know how to fix it, you could create a MR, I would happily review and merge :+1:

**Comment by T0biasCZe** [comment]
> Please send the stacktrace... You know, otherwise I can't help.

i didnt mean any specific error right now, but I meant overall if it stopped later, for example system running out of ram *ehm memory laks*, windows crashing, etc. that would make the conversion get interrupted mid run

**Comment by semjon00** [comment]
@T0biasCZe Sorry, misread your message

**Comment by SauChi666** [comment]
Pls do output files one at a time, not all at the very end,,, so when there is an error I do not have to start all over, I can continue with the files that aare not processed...  it was working like that before... I am having problems with large files...
